aten/src/ATen/native/cuda/SpectralOps.cpp

```diff
@@ -234,6 +234,22 @@ void cufft_clear_plan_cache_impl(int64_t device_index) {
 namespace {
 constexpr int64_t cufft_max_ndim = 3;
 
+// "Large" here means a prime factor not special-cased by cuFFT
+bool has_large_prime_factor(int64_t n) {
+  constexpr int64_t first_large_prime = 11;
+  const std::array<int64_t, 4> prime_radices{{2, 3, 5, 7}};
+  for (auto prime : prime_radices) {
+    if (n < first_large_prime) {
+        return false;
+    }
+
+    while (n % prime == 0) {
```

Comment:
Nit (doesn't have to be changed): In general compilers do a good job of optimizing modulus operations followed by division but this might actually be a case for std::div

Reply:
`std::div` is often slower than just letting the compiler optimize it, since `std::div` is behind a library function call.